File: src/stan-users-guide/regression_1/regression_1.R
#library("rstan")
library("cmdstanr")
set_cmdstan_path("~/cmdstan-2.22.1/") # set to your local install of cmdstan

#Run from command line: Rscript regression_1.R
#If running from RStudio remember to set the working directory 
# >Session>Set Working Directory>To Source File Location

#generate data
alpha <- 2
beta <- 3
sigma <- 5
N <- 100
x <- runif(N, 0, 10)
y <- rnorm(N, alpha + beta*x, sigma)
stan_data <- list(N=N, x=x, y=y)

#dump(c('N','x','y'),file="stan_data.Rdump")

model = cmdstan_model('regression_1.stan')

fit <- model$sample(data = stan_data,num_chains=4)

print(fit$summary())

File: src/stan-users-guide/linear_regression_1/regression_1.py
""" Example program that generates data, compiles and runs regression.stan"""
from cmdstanpy import CmdStanModel
import numpy


# Run from command line: Python regression_1.py

ALPHA = 2
BETA = 3
SIGMA = 5
N = 100
X = numpy.random.uniform(size=N)
Y = numpy.random.normal(size=N, loc=ALPHA + BETA * X, scale=SIGMA)

STAN_DATA = {'N': N, 'x': X, 'y': Y}

STAN_PROGRAM = CmdStanModel(stan_file='regression_1.stan')
STAN_PROGRAM.compile()
FIT = STAN_PROGRAM.sample(data=STAN_DATA, output_dir='.')
print(FIT.summary())
